### PR TITLE
Song preview and like bugfixing

### PR DESCRIPTION
**Pull request by Alberto-Cortes**
Songs can now play a preview if available by tapping the card on the recommendation stack. This preview is also played automatically on the SongDetailFragment.
Fixed a bug that occurred when a song was liked, said song was not pushed to Parse, rather than the last song of the stack. This is now working as intended pushing the swiped song.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -2,6 +2,12 @@ plugins {
     id 'com.android.application'
 }
 
+def apikeyPropertiesFile = rootProject.file("apikey.properties")
+def apikeyProperties = new Properties()
+apikeyProperties.load(new FileInputStream(apikeyPropertiesFile))
+
+
+
```
Comment (mbarbagallo) [inline app/build.gradle:10]: nit: dont commit all this whitespace

File: app/build.gradle [hunk 2]
```diff
@@ -13,6 +19,10 @@ android {
         versionCode 1
         versionName "1.0"
 
+        buildConfigField("String", "SPOTIFY_KEY", apikeyProperties['SPOTIFY_KEY'])
+        buildConfigField("String", "PARSE_APP_ID", apikeyProperties['PARSE_APP_ID'])
+        buildConfigField("String", "PARSE_CLIENT_KEY", apikeyProperties['PARSE_CLIENT_KEY'])
+
```
Comment (mbarbagallo) [inline app/build.gradle:25]: finally!

File: app/src/main/java/com/example/photostomusic/fragments/RecommendationFragment.java [hunk 3]
```diff
@@ -249,27 +254,53 @@ public void discarded(int mIndex, int direction) {
...
                                     break;
                                 case 3:
                                     // Accept, send to parse and move to history
                                     Log.d(TAG, "discarded: 3");
-                                    saveSong(songImage, songName, songArtist, songAlbum, songCode, genres, photo);
+                                    try {
+                                        saveSong(songs.getJSONObject(mIndex), genres, photo);
+                                    } catch (JSONException e) {
+                                        e.printStackTrace();
+                                    }
```
Comment (mbarbagallo) [inline app/src/main/java/com/example/photostomusic/fragments/RecommendationFragment.java:282]: I would rearrange your switch statement to minimize code duplication. so something more like
```
case 1:
case 3:
    try {
        saveSong(songs.getJSONObject(mIndex-1), genres, photo);
    } catch (JSONException e) {
        e.printStackTrace();
    }
    return;
case 0:
case 2:
    Log.d(TAG, "Song discarded");
default:
    mp.stop();
    mp.reset();
```

File: app/src/main/java/com/example/photostomusic/fragments/RecommendationFragment.java [hunk 4]
```diff
@@ -249,27 +254,53 @@ public void discarded(int mIndex, int direction) {
...
                                     break;
                             }
                         }
 
                         @Override
-                        // TODO: (STRETCH) Add preview of song with Spotify API
                         public void topCardTapped() {
+                            mp.stop();
+                            mp.reset();
```
Comment (mbarbagallo) [inline app/src/main/java/com/example/photostomusic/fragments/RecommendationFragment.java:290]: also maybe pull these two (stop, reset) into a function to reduce duplicate code

**Review by mbarbagallo — APPROVED** [review]
Let's try to remove a lot of the duplicate code for the next commit

File: app/src/main/java/com/example/photostomusic/fragments/RecommendationFragment.java (hunk 3 above)
Comment (Alberto-Cortes) [reply app/src/main/java/com/example/photostomusic/fragments/RecommendationFragment.java:282]: I had no idea this was possible! It has now been changed :)
Thanks for the tip!

File: app/src/main/java/com/example/photostomusic/fragments/RecommendationFragment.java (hunk 4 above)
Comment (Alberto-Cortes) [reply app/src/main/java/com/example/photostomusic/fragments/RecommendationFragment.java:290]: Done :)

**Pull request merged**